src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FirstHop.tsx

```diff
@@ -1,89 +1,181 @@
   selectHopTotalProtocolFeesFiatPrecision,
   selectQuoteDonationAmountUsd,
 } from 'state/slices/tradeQuoteSlice/selectors'
-import { store, useAppSelector } from 'state/store'
+import { store, useAppDispatch, useAppSelector } from 'state/store'
 
-import { useTradeExecutooor } from '../hooks/useTradeExecutor'
 import { TradeType } from '../types'
 import { Hop } from './Hop'
 import {
   getApprovalStep,
   getAssetSummaryStep,
   getDonationSummaryStep,
   getHopSummaryStep,
-  getTitleStep,
-  getTradeStep,
 } from './steps'
 
+const useMockAllowanceApproval = (_tradeQuoteStep: TradeQuoteStep, _isExactAllowance: boolean) => {
+  const [approvalTxId, setApprovalTxId] = useState<string>()
+  const [approvalTxStatus, setApprovalTxStatus] = useState<TxStatus>(TxStatus.Unknown)
+  const executeAllowanceApproval = useCallback(() => {
+    setApprovalTxStatus(TxStatus.Pending)
+    const promise = new Promise((resolve, _reject) => {
+      setTimeout(() => setApprovalTxId('0x12345678901234567890'), 2000)
+      setTimeout(() => {
+        setApprovalTxStatus(TxStatus.Confirmed)
+        resolve(undefined)
+      }, 5000)
+    })
+
+    return promise
+  }, [])
+
+  return {
+    isApprovalNeeded: true,
+    executeAllowanceApproval,
+    approvalTxId,
+    approvalTxStatus,
+    approvalNetworkFeeCryptoBaseUnit: '12345678901234',
+  }
+}
+
+const useMockTradeExecution = () => {
+  const [tradeStatus, setTradeStatus] = useState(TxStatus.Unknown)
+  const [sellTxHash, setSellTxHash] = useState<string>()
+  const executeTrade = useCallback(() => {
+    const promise = new Promise((resolve, _reject) => {
+      setTradeStatus(TxStatus.Pending)
+      setTimeout(() => setSellTxHash('0x12345678901234567890'), 2000)
+      setTimeout(() => {
+        setTradeStatus(TxStatus.Confirmed)
+        resolve(undefined)
+      }, 5000)
+    })
+
+    return promise
+  }, [])
+
+  return {
+    buyTxHash: undefined,
+    sellTxHash,
+    tradeStatus,
+    executeTrade,
+  }
+}
+
 export const FirstHop = ({
   swapperName,
   tradeQuoteStep,
   isMultiHopTrade,
+  isOpen,
+  onToggleIsOpen,
 }: {
   swapperName: SwapperName
   tradeQuoteStep: TradeQuoteStep
   isMultiHopTrade: boolean
+  isOpen: boolean
+  onToggleIsOpen: () => void
 }) => {
   const {
     number: { toCrypto, toFiat },
   } = useLocaleFormatter()
   const translate = useTranslate()
-
+  const dispatch = useAppDispatch()
   const [isExactAllowance, toggleIsExactAllowance] = useToggle(false)
   const donationAmountUsd = useAppSelector(selectQuoteDonationAmountUsd)
   const {
-    // TODO: use these
-    // executeTrade,
-    // buyTxHash,
+    // TODO: use the message to better ux
     // message,
+    buyTxHash,
     sellTxHash,
     tradeStatus,
-  } = useTradeExecution()
+    executeTrade,
+  } = useMockTradeExecution() // TODO: use the real hook here
 
   // TODO: use `isApprovalNeeded === undefined` here to display placeholder loading during initial approval check
```

Comment:
Precisely this - or alternatively, a loading state for `isApprovalNeeded` ?

Reply:
yeah will implement in follow-up pr